src/gluonts/mx/trainer/_base.py

```diff
@@ -307,6 +337,25 @@ def loop(
                             epoch_no, validation_iter, is_training=False
                         )
 
+                    # update average trigger
+                    if isinstance(
+                        self.avg_strategy, IterationAveragingStrategy
+                    ):
+                        if isinstance(self.avg_strategy, Alpha_Suffix):
+                            # alpha suffix
+                            self.avg_strategy.update_average_trigger(
+                                epoch_no + 1
+                            )
+                        elif isinstance(self.avg_strategy, (NTA_V1, NTA_V2)):
+                            # NTA
+                            self.avg_strategy.update_average_trigger(
+                                loss_value(epoch_loss)
+                            )
```

Comment:
This is not scalable in case we want to include other strategies or other variations. I think we should have a contract for this method that is general enough to deal with both cases. Maybe something like:
```
self.avg_strategy.update_average_trigger(
        metric=loss_value(epoch_loss),
        epoch=epoch_no + 1
                            )
```
Each strategy can use whatever it needs and can have a `**kwargs` argument for the ones that are not used.